hotel_reservation/models/hotel_quick_reservation.py

```diff
@@ -0,0 +1,118 @@
+# See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class QuickRoomReservation(models.TransientModel):
+    _name = "quick.room.reservation"
+    _description = "Quick Room Reservation"
+
+    partner_id = fields.Many2one("res.partner", "Customer", required=True)
+    check_in = fields.Datetime("Check In", required=True)
+    check_out = fields.Datetime("Check Out", required=True)
+    room_id = fields.Many2one("hotel.room", "Room", required=True)
+    warehouse_id = fields.Many2one("stock.warehouse", "Hotel", required=True)
+    pricelist_id = fields.Many2one("product.pricelist", "pricelist")
+    partner_invoice_id = fields.Many2one(
+        "res.partner", "Invoice Address", required=True
+    )
+    partner_order_id = fields.Many2one("res.partner", "Ordering Contact", required=True)
+    partner_shipping_id = fields.Many2one(
+        "res.partner", "Delivery Address", required=True
+    )
+    adults = fields.Integer("Adults")
+
+    @api.onchange("check_out", "check_in")
+    def on_change_check_out(self):
+        """
+        When you change checkout or checkin it will check whether
+        Checkout date should be greater than Checkin date
+        and update dummy field
+        -----------------------------------------------------------
+        @param self: object pointer
+        @return: raise warning depending on the validation
+        """
+        if self.check_out and self.check_in:
+            if self.check_out < self.check_in:
```

Comment:
Combine both condition
```suggestion
            if (self.check_out and self.check_in) and (self.check_out < self.check_in):
```